pkg/mdns/server/server.go

```diff
@@ -97,9 +97,7 @@ func (s *Server) handlemDNSPacket(conn *net.UDPConn, packet []byte, from net.Add
 
 	// Handle all the questions and construct the answers
 	for _, q := range query.Question {
-		for _, a := range s.responder.Answer(q) {
-			answers = append(answers, a)
-		}
+		answers = append(answers, s.responder.Answer(q)...)
```

Comment:
What happens if s.responder.Answer(q) is empty? Is that possible?

Reply:
Nothing is appended, if `s.responder.Answer(q)` is empty. Usually you have a for loop that iterates over the args of the spread operator. If the args are empty, usually the for loop doesn't get executed.

Experiment:
https://go.dev/play/p/Xfxh9XogEfL